arch/x86/slide_hash_sse2.c

```diff
@@ -12,27 +12,50 @@
 #include "../../deflate.h"
 
 #include <immintrin.h>
+#include <assert.h>
+
+static inline void slide_hash_chain(Pos *table0, Pos *table1, uint32_t entries0,
+                                    uint32_t entries1, const __m128i wsize) {
+    int onChain = 0;
+    Pos *table;
+    uint32_t entries;
+
+next_chain:
+    table = (onChain) ? table1 : table0;
+    entries = (onChain) ? entries1 : entries0;
 
-static inline void slide_hash_chain(Pos *table, uint32_t entries, const __m128i wsize) {
     table += entries;
-    table -= 8;
+    table -= 16;
+    __m128i value0, value1, result0, result1;
 
+    /* ZALLOC allocates this pointer unless the user chose a custom allocator.
+     * Our alloc function is aligned to 64 byte boundaries */
     do {
-        __m128i value, result;
-
-        value = _mm_loadu_si128((__m128i *)table);
-        result= _mm_subs_epu16(value, wsize);
-        _mm_storeu_si128((__m128i *)table, result);
+        value0 = _mm_load_si128((__m128i *)table);
+        value1 = _mm_load_si128((__m128i *)(table + 8));
+        result0 = _mm_subs_epu16(value0, wsize);
+        result1 = _mm_subs_epu16(value1, wsize);
+        _mm_store_si128((__m128i *)table, result0);
+        _mm_store_si128((__m128i *)(table + 8), result1);
 
-        table -= 8;
-        entries -= 8;
+        table -= 16;
+        entries -= 16;
     } while (entries > 0);
+
+    ++onChain;
```

Comment:
Camel case variables..
Code looks pretty clean otherwise.

Reply:
Going to need to write a git hook for this or something when operating on this repo.